content/blog/2023/cilogon-integration/index.md

```diff
@@ -0,0 +1,52 @@
+---
+title: "CILogon usage at 2i2c"
+subtitle: ""
+summary: "The following is a summary of how CILogon is used at 2i2c, how the integration works and a celebration of the partnership."
+authors: ["Georgiana Dolocan"]
+tags: []
+categories: [engineering, partnerships]
+date: "2023-02-23"
+featured: false
+draft: false
+---
+
+## About CILogon
+
+[CILogon](https://www.cilogon.org) is an open source service provider that allows users to log in against over 4000 various identity providers, including campus identity providers. The available identity providers are members of [InCommon](https://incommon.org/federation/), a federation of universities and other organizations that provide single sign-on access to various resources.
+
+## CILogon and 2i2c
+
+For the past year, 2i2c has been successfully using CILogon for more than fifteen of the hubs it manages.
+
+Currently, most of the hubs that use it are hubs for communities in education that want to manage their hub access through their own institutional providers.
+
+With using a tool like CILogon, we allow hub access to be managed both through the communities' institutional providers, but also through social providers like GitHub and Google. Because both authentication mechanisms can coexist, there's no need to provide specific credentials for 2i2c staff in order to have access to the hub. This reduces both the burden on institution's IT departments, but also the complexity of a hub deployment.
+
+Moreover, as we migrate away from our current Auth0 setup, the number of hubs using CILogon will further increase in the following year.
+
+## The setup
+
+The setup that 2i2c uses, is based on two important tools, the CILogon administrative client and the JupyterHub CILogonOAuthenticator.
+
+### The CILogon administrative client
+
+The [2i2c administrative client](https://cilogon.github.io/oa4mp/server/manuals/dynamic-client-registration.html) provided by CILogon allowed us to automatically manage the CILogon OAuth applications needed for authenticating into the hub.
+
+For each hub that uses CILogon, we dynamically create an OAuth [client application](https://cilogon.github.io/oa4mp/server/manuals/dynamic-client-registration.html) in CILogon and store the credentials safely, using the script at [cilogon_app.py](https://github.com/2i2c-org/infrastructure/blob/HEAD/deployer/cilogon_app.py). The script can also used for `updating` the callback URLs of an existing OAuth application, `deleting` a CILogon OAuth application when a hub is removed or changes authentication methods, `getting` details about an existing OAuth application, `getting all` existing 2i2c CILogon OAuth applications.
```

Comment:
The link isnt a permalink but uses HEAD, and may break over time, go for a permalink here.

Reply:
👍🏼 thanks @consideRatio